### ISSUE TITLE
Compile official nginx build 1.15.9 with session_sticky module

### ISSUE THREAD

**Issue by AndrewDi** [alibaba/tengine#1197]
I download official nginx 1.15.9 build, and use add-module with tengine's https://github.com/alibaba/tengine/tree/master/modules/ngx_http_upstream_session_sticky_module module, but get following error:
```c
cc -c -pipe  -O -W -Wall -Wpointer-arith -Wno-unused-parameter -Werror -g   -I src/core -I src/event -I src/event/modules -I src/os/unix -I objs -I src/http -I src/http/modules \
	-o objs/addon/ngx_http_upstream_session_sticky_module/ngx_http_upstream_session_sticky_module.o \
	../ngx_http_upstream_session_sticky_module/ngx_http_upstream_session_sticky_module.c
../ngx_http_upstream_session_sticky_module/ngx_http_upstream_session_sticky_module.c: 在函数‘ngx_http_upstream_session_sticky_init_upstream’中:
../ngx_http_upstream_session_sticky_module/ngx_http_upstream_session_sticky_module.c:1366:21: 错误：‘ngx_http_upstream_rr_peer_t’没有名为‘id’的成员
             if (peer->id.len == 0) {
                     ^
../ngx_http_upstream_session_sticky_module/ngx_http_upstream_session_sticky_module.c:1372:44: 错误：‘ngx_http_upstream_rr_peer_t’没有名为‘id’的成员
             sscf->server[i].sid.data = peer->id.data;
                                            ^
../ngx_http_upstream_session_sticky_module/ngx_http_upstream_session_sticky_module.c:1373:43: 错误：‘ngx_http_upstream_rr_peer_t’没有名为‘id’的成员
             sscf->server[i].sid.len = peer->id.len;
                                           ^
make[1]: *** [objs/addon/ngx_http_upstream_session_sticky_module/ngx_http_upstream_session_sticky_module.o] 错误 1
make[1]: 离开目录“/home/parallels/nginx-1.15.9”
make: *** [build] 错误 2
```
can I get any help to fix this problem here? @wangfakang

**Comment by wangfakang** [comment]
@AndrewDi  Because the module relies on the modification of core code, you can refer to the https://github.com/alibaba/tengine/pull/1112/commits/e568c2d45feaf58bc83e497e42fe9846bce55512 .

BTW we have merged nginx 1.15.9 into latest master for tengine , you can try it.   https://github.com/alibaba/tengine/issues/1196

**Comment by AndrewDi** [comment]
@wangfakang Thanks, I modify `ngx_http_upstream_round_robin` and `ngx_http_upstream`, so that official nginx build can support `ngx_http_upstream_session_sticky_module`.
